conf/machine/apalis-imx8x-smartracks.conf

```diff
@@ -0,0 +1,73 @@
+#@TYPE: Machine
+#@NAME: Toradex Apalis iMX8X
+#@DESCRIPTION: Toradex Apalis iMX8X powered by a i.MX 8X SoC
+#@MAINTAINER: Philippe Schenker <philippe.schenker@toradex.com>
+
+MACHINE_NAME = "Apalis-iMX8X"
+
+# for C0 silicon add mx8qxpc0 to the right of mx8qxp
+SILICON-OVERRIDES ?= "mx8qxpc0:"
+MACHINEOVERRIDES =. "mx8:mx8x:mx8qxp:${SILICON-OVERRIDES}"
+
+IMX_DEFAULT_BSP = "nxp"
+
+require conf/machine/include/imx-base.inc
+# if one wants cortexa35-crypto, set DEFAULTTUNE accordingly
+DEFAULTTUNE_mx8qxp = "aarch64"
+require conf/machine/include/tune-cortexa35.inc
+
+# Don't include kernels in standard images
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base = ""
+
+LOADADDR = ""
+
+# we do not want to have getty running on tty1 as we run
+# auto-serial-console there
+USE_VT = "0"
+
+KERNEL_DEVICETREE = " \
+    freescale/imx8qxp-apalis-eval.dtb \
```

Comment:
You are correct. By right, if we are creating a new board for the Apalis-iMX8X, we should create a new board level device tree. It would be structurally similar to the [eval board device tree](https://git.toradex.com/cgit/linux-toradex.git/tree/arch/arm64/boot/dts/freescale/imx8qxp-apalis-eval.dts?h=toradex_5.4-2.1.x-imx), which includes the [imx8qxp-apalis.dtsi](https://git.toradex.com/cgit/linux-toradex.git/tree/arch/arm64/boot/dts/freescale/imx8qxp-apalis.dtsi?h=toradex_5.4-2.1.x-imx) containing SOM level declarations. However, the issue is that some I2C and GPIO pin configurations we need to customize according to our custom board pinout design is declared in this SOM level device tree (under the iomuxc node) and not the board level device tree. For this use case,  most Toradex documentation suggests using overlays while leaving the board level device tree intact. 

I think we will eventually create our own board level device tree to replace the eval board device tree, though in the meantime we probably will continue to use the overlay for pinmuxing while we figure out how to do that.

Reply:
Thanks for explaining, Yan Huan. In this case, can you help to create a "Technical Debt" work item under "Project Groundwork" feature in AzDO to track it? I agree with you that we can revisit this later.